test/functional/api/vim_spec.lua

```diff
@@ -1036,5 +1036,29 @@ describe('api', function()
       eq(expected, nvim("list_uis"))
     end)
   end)
-
+  describe('nvim_get_keyboard_mods_state', function()
+    it('returs a dictionary on or the specific error', function()
+        -- Returns the dictionary on success otherwise a specified error
+        -- if the resources were not present on the system.
+
+        local status, rv = pcall(meths.get_keyboard_mods_state)
+        if status then
+          if rv['CapsLock'] ~= nil then
+            eq(false, rv['CapsLock'])
```

Comment:
This does not check for keys being boolean, it checks for keys being false. Also do not use three copy-pasted `if` blocks, it is far better with `for _, k in ipairs({'CapsLock', …})`.

Reply:
OK, I will do it, within an hour

Reply:
Though actually no, the other way around:

```Lua
for k, v in pairs(rv) do
  eq(true, {CapsLock=true, …})[k])  -- Check that key is one of the expected keys
  eq('boolean', type(v))
endfor
```